engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -2842,12 +2860,20 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
       }
 
       if (commandLineInfo.args.length > 0) {
-        String filename = expandFileName(commandLineInfo.args[0], true);
+        URI uri = new URI(commandLineInfo.args[0]);
+        final String scheme = uri.getScheme();
+        String filename = uri.getRawPath();
 
         // see if the first argument is actually an app package
         try {
-          AppPackage ap = new AppPackage(new File(filename));
-          ap.close();
+          if (scheme == null || scheme.equals("file")) {
+            filename = expandFileName(uri.getPath(), true);
+            new AppPackage(new FileInputStream(new File(filename)), true).close();
+          } else {
+            try (FileSystem fs = FileSystem.newInstance(uri, conf)) {
+              new AppPackage(fs.open(new Path(filename)), true).close();
+            }
+          }
```

Comment:
Sorry, I assumed the boolean parameter passed to fs.open() is getting passed to AppPackage(). My proposal was having AppPackage(URI uri)